rust/item/src/lib.rs

```diff
@@ -404,6 +404,7 @@ pub mod item {
         msg!("store_mint");
         if store_mint {
             item_class.mint = Some(item_mint.key());
+            item_class.index = class_index;
```

Comment:
You want to add this property to the struct definition also, and update the account storage size.